### PR TITLE
CON-2922 add portals to category page

### PR DESCRIPTION
**Pull request by devbucket**

### CODE REVIEW

File: components/ProductGrid/components/Item/index.jsx [hunk 1]
```diff
@@ -37,59 +37,91 @@ const Item = ({ product, display, isFavorite }) => (
...
         {(!display || display.price) && (
-          <Grid>
-            {product.price.info && (
-              <Grid.Item>
-                <PriceInfo className={styles.basicPrice} text={product.price.info} />
-              </Grid.Item>
-            )}
-          </Grid>
+          <Fragment>
+            <Portal name={portals.PRODUCT_ITEM_PRICE_BEFORE} props={{ productId: product.id }} />
```
Comment (Carsten89) [inline components/ProductGrid/components/Item/index.jsx]: There are no PRODUCT_ITEM_PRICE_XXX constants in common-commerce

File: components/CategoryList/index.jsx [hunk 2]
```diff
@@ -8,6 +8,8 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { bin2hex } from '@shopgate/pwa-common/helpers/data';
+import Portal from '@shopgate/pwa-common/components/Portal';
+import * as portals from '@shopgate/pwa-common-commerce/category/constants/Portals';
```
[the remark this answers is not in the document]
Comment (devbucket) [reply components/CategoryList/index.jsx]: No, we did it like this everywhere with the portals. And in this case the exposed module will be used in all the other places either way. So the output bundle is not affected. I'll keep it like this to stay consistent.

File: components/CategoryList/index.jsx [hunk 3]
```diff
@@ -23,13 +25,18 @@ const CategoryList = ({ categories }) => {
...
           key={category.id}
-          link={`/category/${bin2hex(category.id)}`}
-          title={category.name}
-        />
-      )}
+          name={portals.CATEGORY_LIST_ITEM}
+          props={{ categoryId: category.id }}
+        >
+          <List.Item
+            link={`/category/${bin2hex(category.id)}`}
```
[the remark this answers is not in the document]
Comment (devbucket) [reply components/CategoryList/index.jsx]: Done

File: pages/Category/components/Products/index.jsx [hunk 4]
```diff
@@ -26,7 +26,7 @@ class Products extends Component {
   };
 
   static defaultProps = {
-    getProducts: () => {},
+    getProducts: () => { },
```
[the remark this answers is not in the document]
Comment (devbucket) [reply pages/Category/components/Products/index.jsx]: No. Removed

File: components/ProductGrid/components/Item/index.jsx (hunk 1 above)
Comment (devbucket) [reply components/ProductGrid/components/Item/index.jsx]: Now there are

**Pull request merged**